src/sentry/lang/native/processing.py

```diff
@@ -283,6 +290,32 @@ def _handles_frame(data, frame):
     return is_native_platform(platform) and "instruction_addr" in frame
 
 
+def _expand_addr_mode(addr_mode, modules):
+    if addr_mode in (None, "abs"):
+        return "abs"
+    if addr_mode.startswith("rel:"):
+        arg = addr_mode[4:]
+
+        for idx, module in enumerate(modules):
+            if module.get("debug_id") == arg:
```

Comment:
We've seen crashes with 700 modules and 20k+ frames. It's probably not a huge performance gain, but I'd go with a map here.